core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -581,6 +581,23 @@ class CityConstructions : IsPartOfGameInfoSerialization {
         updateUniques()
     }
 
+    fun removeBuildings(buildings: Set<Building>) {
```

Comment:
This is too complex - filter out builtBuildings, and then call setTransients
Should be 3 lines of code total
- convert to set of names
- removeif
- setTransients